src/lookup_service.cpp

```diff
@@ -101,6 +103,64 @@ LookupService::registerData(const DmtcpMessage &msg, const void *data)
   addKeyValue(msg.nsid, key, keyLen, val, valLen);
 }
 
+void
+LookupService::get64(jalib::JSocket &remote,
+                     const DmtcpMessage &msg)
+{
+  KeyValueMap64 &kvmap = _maps64[msg.kvdbId];
+
+  if (kvmap.find(msg.kvdb.key) == kvmap.end()) {
+    JTRACE("Lookup Failed, Key not found.");
+    remote << DmtcpMessage(DMT_KVDB64_GET_FAILED);
+  }
+
+  DmtcpMessage reply(DMT_KVDB64_GET_RESPONSE);
+  reply.kvdb.value = kvmap[msg.kvdb.key];
```

Comment:
Will the code try to send another message to the remote (line `117-120`) after sending a failed message in the line `114`? Or does a reply to remote work as a `return`?

Reply:
Good catch, @JainTwinkle !